crates/api_crud/src/site/update.rs

```diff
@@ -208,3 +183,201 @@ impl PerformCrud for EditSite {
     Ok(res)
   }
 }
+
+fn validate_update_payload(
+  site_regex: Option<String>,
+  federation_enabled: bool,
+  private_instance: bool,
+  edit_site: &EditSite,
+) -> LemmyResult<()> {
+  // Check that the slur regex compiles, and return the regex if valid...
+  let slur_regex = build_and_check_regex(&site_regex.as_deref())?;
+
+  if let Some(name) = &edit_site.name {
+    // The name doesn't need to be updated, but if provided it cannot be blanked out...
+    site_name_length_check(name)?;
+    check_slurs_opt(&edit_site.name, &slur_regex)?;
+  }
+
+  if let Some(desc) = &edit_site.description {
+    site_description_length_check(desc)?;
+    check_slurs_opt(&edit_site.description, &slur_regex)?;
+  }
+
+  if let Some(listing_type) = &edit_site.default_post_listing_type {
+    // Only allow all or local as default listing types...
+    if listing_type != &ListingType::All && listing_type != &ListingType::Local {
+      return Err(LemmyError::from_message(
+        "invalid_default_post_listing_type",
+      ));
+    }
+  }
+
+  let enabled_private_instance_with_federation = edit_site.private_instance == Some(true)
+    && edit_site.federation_enabled.unwrap_or(federation_enabled);
+  let enabled_federation_with_private_instance = edit_site.federation_enabled == Some(true)
+    && edit_site.private_instance.unwrap_or(private_instance);
+
+  if enabled_private_instance_with_federation || enabled_federation_with_private_instance {
```

Comment:
Same here - should this check also be done when creating a site?

Reply:
Probably a good idea for consistency.